### DISCUSSION TITLE
Function Transformation - Different behaviour to what was shown in November 20th, 2024 Community Call

### DISCUSSION THREAD

**Discussion by rajyraman** [microsoft/semantic-kernel#9800] [Q&A]
I tried out what @markwallace-microsoft showed in the demo and I am getting a different behaviour. Function is not invoked for any questions about colour with auto invocation.

![image](https://github.com/user-attachments/assets/b9db8ba7-1033-451c-b046-10640cb17402)

However, if I force function invocation, it seems to be working (with the exception of the last question).

![image](https://github.com/user-attachments/assets/4e6ceec2-7f08-4153-89c4-70ae6519f71e)

Using gpt-4o-mini and Azure OAI. What could be this issue?

**Comment by rajyraman** [comment]
If I use gpt-4o, I get the expected result. Weird!!

![image](https://github.com/user-attachments/assets/aa25eb8d-5753-436e-9baf-6277384da05d)

**Comment by evchaki** [comment]
@rajyraman - are you using the latest 4o and 4o-mini models?

**Reply by rajyraman** [reply]
@evchaki - It is the latest.
![image](https://github.com/user-attachments/assets/6a730cff-0f6b-44ad-83c6-266d635c6380)

**Comment by evchaki** [comment]
@SergeyMenshykh - this looks like another example of 4o-mini function calling issue.  Can you see if this is related to the other issues you are tagged in.

**Reply by SergeyMenshykh** [reply]
Hi @rajyraman, we will start looking into the issue together with a few other ones related to function calling later this week.

**Comment by SergeyMenshykh** [comment]
Hi @rajyraman, could you please share your plugin so we can see the full picture?
